src/protocols/secure_channel/CASESession.cpp

```diff
@@ -343,37 +360,53 @@ CHIP_ERROR CASESession::SendSigmaR1()
         ReturnErrorCodeIf(mFabricInfo == nullptr, CHIP_ERROR_INCORRECT_STATE);
         memcpy(mIPK, GetIPKList()->data(), sizeof(mIPK));
         ReturnErrorOnFailure(
-            mFabricInfo->GenerateDestinationID(ByteSpan(mIPK), ByteSpan(initiatorRandom), GetPeerNodeId(), destinationIdSpan));
+            mFabricInfo->GenerateDestinationID(ByteSpan(mIPK), ByteSpan(mInitiatorRandom), GetPeerNodeId(), destinationIdSpan));
     }
     ReturnErrorOnFailure(tlvWriter.PutBytes(TLV::ContextTag(3), destinationIdentifier, sizeof(destinationIdentifier)));
 
     ReturnErrorOnFailure(
         tlvWriter.PutBytes(TLV::ContextTag(4), mEphemeralKey.Pubkey(), static_cast<uint32_t>(mEphemeralKey.Pubkey().Length())));
+
+    // If CASE session was previously established using the current state information, let's fill in the session resumption
+    // information in the the Sigma1 request. It'll speed up the session establishment process if the peer can resume the old
+    // session.
+    if (mPairingComplete)
```

Comment:
Does this assumes that we only ever have 1 CASE session established (e.g. single-admin). There is only 1 "mPairingComplete" per CASESession -> do we keep a full CASESession stored/loaded ? I think so... I am trying to understand how we get to here with `mPairingComplete` true/false, and what information is used. It truly needs to be that we are communicating with a node we already had opened CASE. Perhaps `mPairingComplete` could be renamed to `mCaseSessionPreviouslyEstablished` or some similar more accurate name. `Pairing` does not appear in CASE section of spec (or any section of secure channel spec).

Reply:
The CASE session is not currently being stored. My current thinking is this
1. CASEServer (based on device configuration) knows how many CASESessions can be stored in persistent storage for resumption. When a new session is established, it gets the serializable structure for the session, and stores it. If the limit reaches, it can evict the oldest from the storage.
2. On client end, need to build some entity that can store the serializable sessions once it's established. A similar approach can be used as described above.

It makes sense to rename `mPairingComplete`. I'll do that in the next update to the PR.